<?xml version="1.0" ?>

<!-- Not shown, but you can also include test bags to get a testing error!-->
<launch>
	
  <arg name="save_file_path" value="$(find radar_human_tracker)/config/trained_model/TrainBoth/" />

  <arg name="leg_feature_matrix_1" value="$(find radar_human_tracker)/data/leg/indoor_0.xml" />
  <arg name="leg_feature_matrix_2" value="$(find radar_human_tracker)/data/leg/outdoor_0.xml" />
  <arg name="leg_feature_matrix_3" value="$(find radar_human_tracker)/data/leg/0_1_0_0.xml" /> <!--for test-->
  <arg name="leg_feature_matrix_4" value="$(find radar_human_tracker)/data/leg/0_1_1_0.xml" /> <!--for test-->

  <arg name="squat_feature_matrix_1" value="$(find radar_human_tracker)/data/squat/indoor_0.xml" />
  <arg name="squat_feature_matrix_2" value="$(find radar_human_tracker)/data/squat/outdoor_0.xml" />
  <arg name="squat_feature_matrix_3" value="$(find radar_human_tracker)/data/squat/0_1_0_0.xml" /> <!--for test-->
  <arg name="squat_feature_matrix_4" value="$(find radar_human_tracker)/data/squat/0_1_1_0.xml" /> <!--for test-->

  <arg name="neg_feature_matrix_1" value="$(find radar_human_tracker)/data/background/lab_0_0.xml" />
  <arg name="neg_feature_matrix_2" value="$(find radar_human_tracker)/data/background/lab_1_0.xml" />
  <arg name="neg_feature_matrix_3" value="$(find radar_human_tracker)/data/background/lab_2_0.xml" />
  <arg name="neg_feature_matrix_4" value="$(find radar_human_tracker)/data/background/pavement_0_0.xml" />
  <arg name="neg_feature_matrix_5" value="$(find radar_human_tracker)/data/background/pavement_1_0.xml" />
  <arg name="neg_feature_matrix_6" value="$(find radar_human_tracker)/data/background/lab_3_0.xml" />

  <param name="mode" value="0" />   <!--0 for train; 1 for test-->
  <param name="classifier_type" value="opencv_random_forest" />
  <param name="validation_data_ratio" value="0.0" />
  <param name="model_file" value="$(find radar_human_tracker)/config/trained_model/TrainBoth/mlpack_random_forest_0.xml" /> <!--only for test-->
  <param name="feature_set_size" value="0" />
  <param name="undersample_negative_factor" value="20"/>

  <param name="min_points_per_cluster" value="5" />
  <param name="cluster_dist_euclid" value="0.3" />
  <param name="/use_sim_time" value="false"/>
  <param name="range_min" value="0.5"/>
  <param name="range_max" value="8"/>
  <node pkg="radar_human_tracker" type="train_classifier" name="radar_train_classifier" required="true"
        args="
        --leg_features $(arg leg_feature_matrix_1)
        --leg_features $(arg leg_feature_matrix_2)
        --test_leg_features $(arg leg_feature_matrix_3)
        --test_leg_features $(arg leg_feature_matrix_4)

        --squat_features $(arg squat_feature_matrix_1)
        --squat_features $(arg squat_feature_matrix_2)
        --test_squat_features $(arg squat_feature_matrix_3)
        --test_squat_features $(arg squat_feature_matrix_4)

        --test_neg_features $(arg neg_feature_matrix_1)
        --neg_features $(arg neg_feature_matrix_2)
        --neg_features $(arg neg_feature_matrix_6)
        --test_neg_features $(arg neg_feature_matrix_4)
        --neg_features $(arg neg_feature_matrix_5)

              --save_file_path $(arg save_file_path)"
              output="screen">
    <rosparam file="$(find radar_human_tracker)/config/otsu_filter_config.yaml" command="load"/>
  </node>
</launch>


